src/useApiUtil.js

```diff
@@ -33,11 +34,14 @@ const useApiUtil = (term) => {
         
         console.log(term);
 
-        fetchData();
-
+        // Check if the REACT_APP_BACKEND_URL is set
+        if(process.env.REACT_APP_BACKEND_URL){
```

Comment:
move this check at initialization of the component, not just when the fetch happens.